mlflow/tracing/archival/databricks.py

```diff
@@ -0,0 +1,405 @@
+                ) AS assessments
+              FROM
+                assessment_events_parsed
+              GROUP BY
+                trace_id
+            ),
+            -- 5. Optimize tags with window function
+            latest_tags AS (
+              SELECT
+                trace_id,
+                -- Use FIRST_VALUE with proper window frame
+                FIRST_VALUE(body) OVER (
+                    PARTITION BY trace_id
+                    ORDER BY time_unix_nano DESC
+                    ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
+                  ) AS tag_json
+              FROM
+                {events_table}
+              WHERE
+                event_name = 'genai.tags.insert'
+              QUALIFY
+                ROW_NUMBER() OVER (PARTITION BY trace_id ORDER BY time_unix_nano DESC) = 1
+            )
+            -- 6. Main query with optimized structure
+            SELECT
+              rs.trace_id,
+              rs.client_request_id,
+              rs.request_time,
+              rs.state,
+              rs.execution_duration_ms,
+              rs.request,
+              rs.response,
+              rs.trace_metadata,
+              -- Optimized tags processing
+              COALESCE(FROM_JSON(lt.tag_json, 'MAP<STRING, STRING>'), MAP()) AS tags,
+              -- Simplified trace_location construction
+              NAMED_STRUCT(
+                'type',
+                CASE
+                  WHEN rs.experiment_id IS NOT NULL THEN 'mlflow_experiment'
+                  WHEN rs.inference_table_name IS NOT NULL THEN 'inference_table'
+                  ELSE NULL
+                END,
+                'mlflow_experiment',
+                CASE
+                  WHEN rs.experiment_id IS NOT NULL THEN NAMED_STRUCT('experiment_id', rs.experiment_id)
+                  ELSE CAST(NULL AS STRUCT<experiment_id: STRING>)
+                END,
+                'inference_table',
+                CASE
+                  WHEN
+                    rs.inference_table_name IS NOT NULL
+                  THEN
+                    NAMED_STRUCT('full_table_name', rs.inference_table_name)
+                  ELSE CAST(NULL AS STRUCT<full_table_name: STRING>)
+                END
+              ) AS trace_location,
+              COALESCE(aa.assessments, ARRAY()) AS assessments,
+              COALESCE(sa.spans, ARRAY()) AS spans
+            FROM
+              root_spans rs
+                LEFT JOIN latest_tags lt
+                  ON rs.trace_id = lt.trace_id
+                LEFT JOIN assessments_agg aa
+                  ON rs.trace_id = aa.trace_id
+                LEFT JOIN spans_agg sa
+                  ON rs.trace_id = sa.trace_id
+            """
+        
+        spark.sql(query)
+        _logger.info(f"Successfully created trace archival view: {view_name}")
+        
+    except Exception as e:
+        raise MlflowException(f"Failed to create trace archival view {view_name}: {str(e)}") from e
```

Comment:
```suggestion
        raise MlflowException(f"Failed to create trace archival view {view_name}") from e
```
nit: I think we don't need str(e) because `from e` makes it visible in the stack trace.